openedx/core/lib/api/paginators.py

```diff
@@ -0,0 +1,52 @@
+""" Paginatator methods for edX API implementations."""
+
+from django.http import Http404
+from django.utils.translation import ugettext as _
+from django.core.paginator import Paginator, InvalidPage
+
+
+def paginate_search_results(object_class, search_results, page_size, page):
+    """
+    Takes edx-search results and returns a Page object populated
+    with db objects for that page.
+
+    :param object_class: Model class to use when querying the db for objects.
+    :param search_results: edX-search results.
+    :param page_size: Number of results per page.
+    :param page: Page number.
+    :return: Paginator object with model objects
+    """
+    paginator = Paginator(search_results['results'], page_size)
+
+    # This code is taken from within the GenericAPIView#paginate_queryset method.
+    # It is common code, but
+    try:
+        page_number = paginator.validate_number(page)
+    except InvalidPage:
+        if page == 'last':
+            page_number = paginator.num_pages
+        else:
+            raise Http404(_("Page is not 'last', nor can it be converted to an int."))
+
+    try:
+        paged_results = paginator.page(page_number)
+    except InvalidPage as e:  # pylint: disable=invalid-name
+        raise Http404(_('Invalid page (%(page_number)s): %(message)s') % {
```

Comment:
Please use Python-style {name}.format variables for translation purposes.

Also do not translate strings unless they will be directly shown to users.

http://edx.readthedocs.org/projects/edx-developer-guide/en/latest/internationalization/i18n.html#use-named-placeholders